src/ui/pages/Identifiers/Identifiers.tsx

```diff
@@ -116,27 +115,33 @@ const Identifiers = () => {
   }, [currentOperation, history.location.pathname]);
   useEffect(() => {
     setShowPlaceholder(identifiersData.length === 0);
-    setAllIdentifiers(
-      identifiersData
-        .filter((identifier) => !identifier.isPending)
-        .filter((identifier) => !identifier.groupMetadata?.groupId)
-        .filter(
-          (identifier) =>
-            !favouritesIdentifiers?.some((fav) => fav.id === identifier.id)
-        )
-    );
-    setFavIdentifiers(
-      identifiersData.filter((identifier) =>
-        favouritesIdentifiers?.some((fav) => fav.id === identifier.id)
-      )
-    );
-    setPendingIdentifiers(
-      identifiersData.filter((identifier) => identifier.isPending)
-    );
-    setMultiSigIdentifiers(
-      identifiersData.filter((identifier) => identifier.groupMetadata?.groupId)
-    );
-  }, [favouritesIdentifiers, identifiersData, toggleClick]);
+    const tmpPendingIdentifiers = [];
+    const tmpMultisigIdentifiers = [];
+    const tmpFavIdentifiers = [];
+    const tmpAllIdentifiers = [];
+    for (const identifier of identifiersData) {
+      if (identifier.isPending) {
+        tmpPendingIdentifiers.push(identifier);
+      }
+      if (identifier.groupMetadata?.groupId) {
+        tmpMultisigIdentifiers.push(identifier);
+      }
+      if (favouritesIdentifiers?.some((fav) => fav.id === identifier.id)) {
+        tmpFavIdentifiers.push(identifier);
+      }
+      if (
+        !identifier.isPending &&
+        !identifier.groupMetadata?.groupId &&
+        !favouritesIdentifiers?.some((fav) => fav.id === identifier.id)
+      ) {
+        tmpAllIdentifiers.push(identifier);
+      }
```

Comment:
If we have `continue` above - we can remove this if statement and just have `tmpAllIdentifiers.push(identifier);` (since it's the final bucket)